cocotb/share/lib/embed/embed.cpp

```diff
@@ -33,18 +33,22 @@ static bool init_failed = false;
 
 
 #if ! defined(__linux__) && ! defined(__APPLE__)
-HANDLE act_ctx = NULL;
-
-BOOL WINAPI DllMain(HINSTANCE, DWORD fdwReason, LPVOID)
+ACTCTX act_ctx = {
+    .cbSize = sizeof(ACTCTX),
+    .dwFlags = ACTCTX_FLAG_HMODULE_VALID | ACTCTX_FLAG_RESOURCE_NAME_VALID,
+    .lpSource = NULL,
+    .wProcessorArchitecture = 0,
+    .wLangId = 0,
+    .lpAssemblyDirectory = NULL,
+    .lpResourceName = MAKEINTRESOURCE(1000),
+    .lpApplicationName = NULL,
+    .hModule = 0
+};
```

Comment:
Designated initializers in C++ prior to C++20 are a GCC extension. Is there some other way to write this? If not, and MSVC will swallow it, so be it. Also should be `static`.